### PR TITLE
Added ServerID function

### PR DESCRIPTION
**Pull request by ewoutp**
This PR extends #106

### CODE REVIEW

File: agency/lock.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+	// Release the lock
+	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
+	defer cancel()
+	if err := l.api.RemoveKeyIfEqualTo(ctx, l.key, l.id); err != nil {
+		return driver.WithStack(err)
+	}
+
+	// Cleanup
+	l.locked = false
```
Comment (neunhoef) [inline agency/lock.go]: I think here is a data race: You first remove the key in the agency, then set `locked` to false (irrelevant here), and only then stop the renewal. I suggest to first cancel the renewal, make sure it is cancelled (unless `l.cancelRenewal()` does this automatically), then remove the key and only then set `locked` to `false`.

Comment (neunhoef) [reply agency/lock.go]: OK, now I understand, the renewal has a precondition as well. OK, this is good. Nevertheless, I think the change of order makes the code easier understandable. Furthermore, If the `RemoveKeyIfEqualTo` fails, then the renewal go routine might keep running. If we have cancelled first, it is gone in any case.

Comment (ewoutp) [reply agency/lock.go]: Note that this function (and the renewal) first claim an in-memory mutex.
So there can be no race.

Second. If RemoveKeyIfEqualTo fails, the lock is likely still held, so I expect the renewal to continue.

Comment (neunhoef) [reply agency/lock.go]: Agreed.

**Pull request merged**